Gems/LmbrCentral/Code/Tests/ShapeGeometryUtilTest.cpp

```diff
@@ -91,4 +93,43 @@ namespace UnitTest
 
         EXPECT_TRUE(triangles.size() == 18);
     }
+
+    // test double to record if DrawTrianglesIndexed or DrawLines are called
+    class DebugShapeDebugDisplayRequests : public AzFramework::DebugDisplayRequests
+    {
+    public:
+        void DrawTrianglesIndexed(
+            [[maybe_unused]] const AZStd::vector<AZ::Vector3>& vertices,
+            [[maybe_unused]] const AZStd::vector<AZ::u32>& indices,
+            [[maybe_unused]] const AZ::Color& color) override
+        {
+            m_drawTrianglesIndexedCalled = true;
+        }
+
+        void DrawLines([[maybe_unused]] const AZStd::vector<AZ::Vector3>& lines, [[maybe_unused]] const AZ::Color& color) override
+        {
+            m_drawLinesCalled = true;
+        }
+
+        bool m_drawTrianglesIndexedCalled = false;
+        bool m_drawLinesCalled = false;
+    };
+
+    // DrawShape internally calls DrawTrianglesIndexed and DrawLines - with no geometry
+    // we want to make sure the shape is not submitted to be drawn
+    TEST(ShapeGeometry, Shape_not_attempted_to_be_drawn_with_no_geometry)
+    {
+        using ::testing::Eq;
+
+        // given
+        DebugShapeDebugDisplayRequests debugDisplayRequests;
+
+        // when
+        LmbrCentral::DrawShape(
+            debugDisplayRequests, LmbrCentral::ShapeDrawParams{ AZ::Colors::White, AZ::Colors::White, true }, LmbrCentral::ShapeMesh{});
+
+        // then
+        EXPECT_THAT(debugDisplayRequests.m_drawTrianglesIndexedCalled, Eq(false));
+        EXPECT_THAT(debugDisplayRequests.m_drawLinesCalled, Eq(false));
```

Comment:
Why not use EXPECT_FALSE(...)?  Seems a little more succinct than EXPECT_THAT(..., Eq(false)).

Reply:
Hmm I tend to prefer this style (Hamcrest) as it can be used in more complex scenarios and fits the same shape. I also quite like how reading left to right reads more like a normal sentence than `EXPECT_FALSE` so I'm tempted to leave this as-is